eventsources/eventing.go

```diff
@@ -69,88 +70,101 @@ type EventingServer interface {
 }
 
 // GetEventingServers returns the mapping of event source type and list of eventing servers
-func GetEventingServers(eventSource *v1alpha1.EventSource, metrics *eventsourcemetrics.Metrics) map[apicommon.EventSourceType][]EventingServer {
+func GetEventingServers(eventSource *v1alpha1.EventSource, metrics *eventsourcemetrics.Metrics) (map[apicommon.EventSourceType][]EventingServer, map[string]v1alpha1.EventSourceFilter) {
 	result := make(map[apicommon.EventSourceType][]EventingServer)
+	filters := make(map[string]v1alpha1.EventSourceFilter)
 	if len(eventSource.Spec.AMQP) != 0 {
 		servers := []EventingServer{}
 		for k, v := range eventSource.Spec.AMQP {
+			filters[getFilterKey(apicommon.AMQPEvent, k)] = *v.Filter
```

Comment:
If `v.Filter` is `nil`, it will panic.

Reply:
Added guard conditions for all eventSources